### PR TITLE
make some of the TRT attributes optional

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: model/src/main/resources/TestResultsTransmissionFormat.xsd [hunk 1]
```diff
@@ -23,7 +23,7 @@ governing permissions and limitations under the License.
             <xs:attribute name="testId" use="required"/>
             <xs:attribute name="bankKey" type="xs:unsignedInt"/>
             <xs:attribute name="contract"/>
-            <xs:attribute name="mode" use="required">
+            <xs:attribute name="mode" default="online">
```
Comment (malaffoon) [inline model/src/main/resources/TestResultsTransmissionFormat.xsd]: For the enum attributes, i chose to make a default for each.

File: model/src/test/java/org/opentestsystem/rdw/common/model/trt/XmlUtilsTest.java [hunk 2]
```diff
@@ -66,6 +66,27 @@ public void itShouldUnmarshalSamplesFromTDS() {
...
+
+    @Test
+    public void itShouldUnmarshalMinimalSample() {
+        final TDSReport tdsReport = XmlUtils.tdsReportFromXml(this.getClass().getResourceAsStream("/TDSReport.minimal.xml"));
+        assertThat(tdsReport.getTest().getGrade()).isEqualTo("7");
+        // test attributes with default values
+        assertThat(tdsReport.getTest().getMode()).isEqualTo("online");
+        assertThat(tdsReport.getOpportunity().getStatus()).isEqualTo("scored");
+        assertThat(tdsReport.getOpportunity().getValidity()).isEqualTo("valid");
+    }
```
Comment (malaffoon) [inline model/src/test/java/org/opentestsystem/rdw/common/model/trt/XmlUtilsTest.java]: i might come back and assert all the null values but i figured that isn't necessary to move forward

File: model/src/main/resources/TestResultsTransmissionFormat.xsd (hunk 1 above)
Comment (malaffoon) [reply model/src/main/resources/TestResultsTransmissionFormat.xsd]: I will forward these changes to Alex once we vet the entire processing pipeline.

File: model/src/main/java/org/opentestsystem/rdw/common/model/trt/Opportunity.java [hunk 3]
```diff
@@ -497,7 +498,7 @@ public void setCompleteness(String value) {
      *
      * @return opportunity
      */
-    public long getOpportunity() {
+    public Long getOpportunity() {
```
Comment (jtreuting) [inline model/src/main/java/org/opentestsystem/rdw/common/model/trt/Opportunity.java]: is this different than oppId which we said is required?

**Comment by malaffoon** [comment]
I'm going to merge so we can test downstream stuff

**Pull request merged**